src/Tools/dotnet-openapi/src/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,208 @@
+
+        protected abstract Task<int> ExecuteCoreAsync();
+
+        protected virtual bool ValidateArguments()
+        {
+            Ensure.NotNullOrEmpty(SourceFileArg.Value, SourceFileArgName);
+            return true;
+        }
+
+        private async Task<int> ExecuteAsync()
+        {
+            if (!ValidateArguments() || Help.HasValue())
+            {
+                ShowHelp();
+                return 1;
+            }
+
+            return await ExecuteCoreAsync();
+        }
+
+        internal FileInfo ResolveProjectFile(CommandOption projectOption)
+        {
+            string csproj;
+            if (projectOption.HasValue())
+            {
+                csproj = projectOption.Value();
+                csproj = Path.Combine(WorkingDir, csproj);
+                if (!File.Exists(csproj))
+                {
+                    Error.Write("The given csproj does not exist.");
+                }
+            }
+            else
+            {
+                var csprojs = Directory.GetFiles(WorkingDir, "*.csproj", SearchOption.TopDirectoryOnly);
+                if (csprojs.Length == 0)
+                {
+                    Error.Write("No csproj files were found in the current directory. Either move to a new directory or provide the project explicitly");
+                }
+                if (csprojs.Length > 1)
+                {
+                    Error.Write("More than one csproj was found in this directory, either remove a duplicate or explicitly provide the project.");
+                }
+
+                csproj = csprojs.Single();
+            }
+
+            return new FileInfo(csproj);
+        }
+
+        protected Project LoadProject(FileInfo projectFile)
+        {
+            var project = ProjectCollection.GlobalProjectCollection.LoadProject(
+                projectFile.FullName,
+                globalProperties: null,
+                toolsVersion: null);
+            project.ReevaluateIfNecessary();
+            return project;
+        }
+
+        internal bool IsProjectFile(string file)
+        {
+            return File.Exists(file) && file.EndsWith(".csproj");
+        }
+
+        internal bool IsUrl(string file)
+        {
+            return Uri.TryCreate(file, UriKind.Absolute, out var _) && file.StartsWith("http");
+        }
+
+        public async Task DownloadAndOverwriteAsync(string sourceFile, string destinationPath, bool overwrite)
+        {
+            Application parent;
+            if(Parent is Application)
+            {
+                parent = (Application)Parent;
+            }
+            else
+            {
+                parent = (Application)Parent.Parent;
+            }
+
+            var content = await parent.DownloadProvider(sourceFile);
+            await WriteToFile(content, destinationPath, overwrite);
+        }
+
+        private async Task WriteToFile(string content, string destinationPath, bool overwrite)
```

Comment:
Not a fan of serializing the downloaded content to a `string`. Suggest copying the `Stream` to the output file directly